Portal/src/Datahub.Portal/Controllers/MediaController.cs

```diff
@@ -0,0 +1,47 @@
+using Datahub.Application.Configuration;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+
+namespace Datahub.Portal.Controllers;
+
+[ApiController]
+public class MediaController : Controller
+{
+    private readonly DatahubPortalConfiguration _datahubPortalConfiguration;
+
+    public MediaController(DatahubPortalConfiguration datahubPortalConfiguration)
+    {
+        _datahubPortalConfiguration = datahubPortalConfiguration;
+    }
+    
+    /// <summary>
+    /// Redirect the video mp4 to the azure storage blob and return the video stream.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("api/media/{**filePath}")]
+    public async Task<IActionResult> GetMedia(string filePath)
+    {
+        // verify the user is authenticated
+        if (User.Identity is { IsAuthenticated: false })
```

Comment:
I think you don't need this if you decorate the Controller with [Authorize], but as it is works.